From 9b3aaa619622bbad6e24449a1b462923ad252f89 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:11:04 -0400
Subject: [powerpc] modules: use r12 for stub jump address

Message-id: <1410545655-205645-436-git-send-email-dzickus@redhat.com>
Patchwork-id: 94622
O-Subject: [RHEL7 PATCH 435/626] powerpc: modules: use r12 for stub jump address.
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit b1ce369e820aaca3d91e9d9bbaaf860794d9ab01
Author: Rusty Russell <rusty@rustcorp.com.au>
Date:   Tue Mar 18 20:12:59 2014 +1030

    powerpc: modules: use r12 for stub jump address.

    In ELFv2, r12 is supposed to equal to PC on entry to a function.
    Our stubs use r11, so change swap that with r12.

    Signed-off-by: Rusty Russell <rusty@rustcorp.com.au>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/module_64.c b/arch/powerpc/kernel/module_64.c
index 8bfcf1b..f8b6d28 100644
--- a/arch/powerpc/kernel/module_64.c
+++ b/arch/powerpc/kernel/module_64.c
@@ -67,13 +67,13 @@ struct ppc64_stub_entry
    r2) into the stub. */
 static struct ppc64_stub_entry ppc64_stub =
 { .jump = {
- 0x3d820000,   /* addis   r12,r2, <high> */
- 0x398c0000,   /* addi    r12,r12, <low> */
+ 0x3d620000,   /* addis   r11,r2, <high> */
+ 0x396b0000,   /* addi    r11,r11, <low> */
  /* Save current r2 value in magic place on the stack. */
  0xf8410000|R2_STACK_OFFSET, /* std     r2,R2_STACK_OFFSET(r1) */
- 0xe96c0020,   /* ld      r11,32(r12) */
- 0xe84c0028,   /* ld      r2,40(r12) */
- 0x7d6903a6,   /* mtctr   r11 */
+ 0xe98b0020,   /* ld      r12,32(r11) */
+ 0xe84b0026,   /* ld      r2,40(r11) */
+ 0x7d8903a6,   /* mtctr   r12 */
  0x4e800420   /* bctr */
 } };
 
-- 
1.7.1